conans/test/functional/devlayout/devlayout_test.py

```diff
@@ -0,0 +1,307 @@
+            generators = "cmake"
+            exports_sources = "src/*", "CMakeLists.txt"
+            generators = "cmake"
+            layout = "cmake"
+            
+            def build(self):
+                cmake = CMake(self) # Opt-in is defined having toolchain
+                cmake.configure()
+                cmake.build()
+
+            def package(self):
+                self.lyt.package()
+
+            def package_info(self):
+                self.lyt.package_info()
+                self.cpp_info.libs = ["hello"]
+            """)
+    cmake = textwrap.dedent("""
+        cmake_minimum_required(VERSION 2.8.12)
+        set(CMAKE_CXX_COMPILER_WORKS 1)
+        set(CMAKE_CXX_ABI_COMPILED 1)    
+        project(HelloWorldLib CXX)
+        
+        include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+        conan_basic_setup()
+
+        add_library(hello src/hello.cpp)
+        add_executable(app src/app.cpp)
+        target_link_libraries(app PRIVATE hello)
+        """)
+    hellopp = textwrap.dedent("""
+        #include "hello.h"
+
+        std::string hello(){
+            #ifdef 	_M_IX86
+                #ifdef NDEBUG
+                return  "Hello World Release 32bits!";
+                #else
+                return  "Hello World Debug 32bits!";
+                #endif
+            #else
+                #ifdef NDEBUG
+                return  "Hello World Release!";
+                #else
+                return  "Hello World Debug!";
+                #endif
+            #endif
+        }
+        """)
+    helloh = textwrap.dedent("""
+        #pragma once
+        #include <string>
+
+        #ifdef WIN32
+          #define HELLO_EXPORT __declspec(dllexport)
+        #else
+          #define HELLO_EXPORT
+        #endif
+
+        HELLO_EXPORT 
+        std::string hello();
+        """)
+    app = textwrap.dedent(r"""
+        #include <iostream>
+        #include "hello.h"
+
+        int main(){
+            std::cout << "****\n" << hello() << "\n****\n\n";
+        }
+        """)
+    test_cmake = textwrap.dedent("""
+        cmake_minimum_required(VERSION 2.8.12)
+        set(CMAKE_CXX_COMPILER_WORKS 1)
+        set(CMAKE_CXX_ABI_COMPILED 1)
+
+        project(Greet CXX)
+
+        include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+        conan_basic_setup()
+
+        add_executable(app app.cpp)
+        MESSAGE(" CONAN LIBS=> ${CONAN_LIBS}")
+        target_link_libraries(app ${CONAN_LIBS})
+        """)
+    test_conanfile = textwrap.dedent("""
+        import os
+        from conans import ConanFile, CMake, tools
+
+        class Test(ConanFile):
+            settings = "os", "compiler", "build_type", "arch"
+            generators = "cmake"
+
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure()
+                cmake.build()
+            
+            def imports(self):
+                self.copy(pattern="*.dll", dst="bin", src="#bindir")
```

Comment:
This will fail until the other PR is merged, right?

Reply:
Yes. I'll remove it.